src/Control/Algebra.hs

```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE FlexibleContexts, FlexibleInstances, GeneralizedNewtypeDeriving, MultiParamTypeClasses, UndecidableInstances #-}
+module Control.Algebra
+( Effects(..)
+) where
+
+import qualified Control.Applicative as Alternative
+import           Control.Algebra.Class
+import qualified Control.Effect.Choose as Effect
+import qualified Control.Effect.Empty as Effect
+import qualified Control.Effect.Fail as Effect
+import           Control.Monad (MonadPlus(..))
+import qualified Control.Monad.Fail as Fail
+import           Control.Monad.Trans.Class
+
+newtype Effects m a = Effects { runEffects :: m a }
```

Comment:
I vote we go with `Eff`. This is something that the `<|>`-using masses are going to need to type a lot.

Also, some documentation here as to its purpose would be 💯